### ISSUE TITLE
Cannot exec into pods: failed to create runc console socket

### ISSUE THREAD

**Issue by soapergem** [ubuntu/microk8s#358]
I have an Ubuntu Server running MicroK8S, installed via Snap. I'm trying to create a simple pod using `kubectl run` that I can exec into for debugging purposes. But it fails every time.

At first, I tried this command:

```
kubectl run -it --rm --restart=Never busybox --image=busybox -- /bin/ash
```

But every time I try to run that, I get this error:

> pod "busybox deleted"
> pod default/busybox terminated (ContainerCannotRun)
> failed to create OCI runtime control socket: stat /run/user/0/snap.microk8s: no such file or directory: unknown

Then I decided to try and run the container first, and exec into second, as two separate commands. Surprisingly, the initial pod creation actually works, a la this command:

```
kubectl run --image=busybox --restart=Never busybox --image=busybox --command -- tail -f /dev/null
```

But then when I try to exec into it using this command...

```
kubectl exec -it busybox -- /bin/ash
```

...I wind up with this error:

> failed to create runc console socket: stat /run/user/0/snap.microk8s: no such file or directory: unknown
> command terminated with exit code 126

Both errors only come into play once I try to connect to a running pod, and both errors reference `/run/user/0/snap.microk8s`. I found this odd since of course there is no user "0." The default user is 1000. But at the end of the day, I'm not really sure what those errors mean, though. Is that a problem with my configuration? Or am I missing some dependency? Or is the hard disk corrupt? Or something else entirely? And ultimately: how can I get this working?

I've also attached the result of `microk8s.inspect` as requested: [inspection-report-20190314_175203.tar.gz](https://github.com/ubuntu/microk8s/files/2967805/inspection-report-20190314_175203.tar.gz)

**Comment by ktsakalozos** [comment]
Hi @soapergem,

I see you have been running two dockerd running using the same configuration:

```
root     22122     1  0  2018 ?        10:47:31 dockerd -G docker --exec-root=/var/snap/docker/321/run/docker --data-root=/var/snap/docker/common/var-lib-docker --pidfile=/var/snap/docker/321/run/docker.pid --config-file=/var/snap/docker/321/config/daemon.json --debug
root     26038     1  3 12:39 ?        00:10:41 /snap/microk8s/460/usr/bin/dockerd -H unix:///var/snap/microk8s/460/docker.sock --exec-root /var/snap/microk8s/common/var/run/docker --graph /var/snap/microk8s/common/var/lib/docker --pidfile /var/snap/microk8s/common/docker-pid --config-file=/var/snap/microk8s/460/args/docker-daemon.json
```

This is setup of running two dockerds is not supported (https://github.com/ubuntu/microk8s/blob/master/README.md#user-guide) because the two dockerd fight over ownership of the running containers. You can see this from the denials reported by apparmor in dmesg.

We will attempt to lift the constraint of not running a second dockerd by moving to containerd in the upcomming 1.14 release. You can try the new release out with:
```
sudo snap install microk8s --classic --channel=1.14/beta
```

Let me know how that went. Thanks

**Comment by soapergem** [comment]
I don't remember making any conscious decision to run two separate instances of dockerd; I simply installed Ubuntu Server 18.04 and chose microk8s from the list of software to install. It's possible though that perhaps I also picked Docker from that list? In any case, I ended up reformatting and reinstalling Ubuntu Server from scratch and this problem went away. I will be filing a new issue shortly however as I have been experiencing all new problems. Thanks for your help.
